drivers/serial/uart_cmsdk_apb.c

```diff
@@ -259,6 +260,13 @@ static int uart_cmsdk_apb_fifo_read(struct device *dev,
 static void uart_cmsdk_apb_irq_tx_enable(struct device *dev)
 {
 	UART_STRUCT(dev)->ctrl |= UART_TX_IN_EN;
+	/* The expectation is TX is a level interrupt, active for as long
+	 * as TX buffer is empty. But in CMSDK UART, it appears to be edge
+	 * interrupt, firing on a state change of TX buffer. So, we need
+	 * to "prime" it here by calling ISR directly, to get interrupt
+	 * processing going.
+	 */
+	uart_cmsdk_apb_isr(dev);
```

Comment:
Please correct me if I am wrong.
This will call interrupt service routine, interrupt itself will be not triggered ?

Reply:
Sure, that's correct.

Reply:
Ok, it is a bit confusing to me.  Let it be.